server/src/main/java/io/crate/expression/operator/any/AnyNotLikeOperator.java

```diff
@@ -97,4 +97,11 @@ protected Query literalMatchesAnyArrayRef(Function any, Literal<?> probe, Refere
             RegexpFlag.COMPLEMENT.value()
         );
     }
+
+    @Override
+    protected void validateRightArg(String arg) {
+        if (arg.endsWith(LikeOperators.DEFAULT_ESCAPE_STR)) {
+            throw new IllegalArgumentException("pattern must not end with escape character");
```

Comment:
I think this would be a more useful error message if it included the arg somewhere?